diagnostics/forcing_feedback/forcing_feedback_kernelcalcs.py

```diff
@@ -0,0 +1,337 @@
+
+
+
+varnames = ["ta","ts","hus","rsus","rsds","rsuscs","rsdscs","rsdt","rsut","rsutcs","rlut","rlutcs"]
+model_mainvar_pert = {}
+i=0
+for varname in varnames:
+    nc = xr.open_dataset(os.environ["DATADIR"]+"/mon/"+os.environ[varname+"_file"])
+    model_mainvar_pert[os.environ[varname+"_var"]] = nc[os.environ[varname+"_var"]].values
+    if i == 0:
+       lat_model = nc[os.environ["lat_coord"]].values
+       lon_model = nc[os.environ["lon_coord"]].values
+       lev_model = nc[os.environ["lev_coord"]].values
+       time_model = nc[os.environ["time_coord"]].values
+    nc.close()
+    i += 1
+i = None
+
+model_mainvar_climo = model_mainvar_pert
+
+#To avoid issues with interpreting different model time formats, just create new variable. 
+time_model = np.arange(len(time_model))+1
+
+#Regrid kernels using function to match horizontal grid of model/observational data
+lw_t_kern_TOA = latlonr3_3D4D(lw_t_kern_TOA,lat_kern,lon_kern,lat_model,lon_model, \
+             model_mainvar_climo[os.environ["ta_var"]])
+lwclr_t_kern_TOA = latlonr3_3D4D(lwclr_t_kern_TOA,lat_kern,lon_kern,lat_model,lon_model, \
+             model_mainvar_climo[os.environ["ta_var"]])
+lw_ts_kern_TOA = latlonr3_3D4D(lw_ts_kern_TOA,lat_kern,lon_kern,lat_model,lon_model, \
+             model_mainvar_climo[os.environ["ts_var"]])
+lwclr_ts_kern_TOA = latlonr3_3D4D(lwclr_ts_kern_TOA,lat_kern,lon_kern,lat_model,lon_model, \
+             model_mainvar_climo[os.environ["ts_var"]])
+lw_q_kern_TOA = latlonr3_3D4D(lw_q_kern_TOA,lat_kern,lon_kern,lat_model,lon_model, \
+             model_mainvar_climo[os.environ["ta_var"]])
+lwclr_q_kern_TOA = latlonr3_3D4D(lwclr_q_kern_TOA,lat_kern,lon_kern,lat_model,lon_model, \
+             model_mainvar_climo[os.environ["ta_var"]])
+sw_q_kern_TOA = latlonr3_3D4D(sw_q_kern_TOA,lat_kern,lon_kern,lat_model,lon_model, \
+             model_mainvar_climo[os.environ["ta_var"]])
+swclr_q_kern_TOA = latlonr3_3D4D(swclr_q_kern_TOA,lat_kern,lon_kern,lat_model,lon_model, \
+             model_mainvar_climo[os.environ["ta_var"]])
+sw_a_kern_TOA = latlonr3_3D4D(sw_a_kern_TOA,lat_kern,lon_kern,lat_model,lon_model, \
+             model_mainvar_climo[os.environ["ts_var"]])
+swclr_a_kern_TOA = latlonr3_3D4D(swclr_a_kern_TOA,lat_kern,lon_kern,lat_model,lon_model, \
+             model_mainvar_climo[os.environ["ts_var"]])
+ps_kern_TOA = latlonr3_3D4D(ps_kern_TOA,lat_kern,lon_kern,lat_model,lon_model, \
+             model_mainvar_climo[os.environ["ts_var"]])
+
+
+#Kernels have now been regridded to the model/observation grid
+lat_kern = lat_model
+lon_kern = lon_model
+
+#If necessary, conduct vertical interpolation
+
+for varname in varnames:
+    if len(model_mainvar_pert[os.environ[varname+"_var"]].shape)==4:
+       #If vertical coordinates for kernel and model data differ, check if just a difference in units,
+       #check if one is flipped and finally, flip or interpolate as needed.
```

Comment:
hPa vs. Pa unit conversion should be handled by specifying which units are desired in the Z axis entry in the settings file (see [documentation](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/ref_settings.html#z-axis-height-depth-pressure)). Interpolation should be retained.